src/js/controller/model.js

```diff
@@ -219,6 +220,10 @@ define([
             _provider.volume(_this.get('volume'));
             _provider.mute(_this.get('mute'));
             _provider.on('all', _videoEventHandler, this);
+
+            if (this.get('instreamMode') === true) {
+                _provider.instreamMode = true;
```

Comment:
Took a look, and still seems like this would be the best place to contain both the setting and removal of the `instreamMode` flag